### PR TITLE
MAINT, REL: Prepare for SciPy 1.13.0 "final" (proposing to skip RC2 for Numpy 2 series support)

### PR DESCRIPTION
**Pull request by tylerjereddy**
This PR contains a number of backports outlined below, and tentatively aims to position us to skip RC2 and just right to SciPy `1.13.0` "final" because of the urgency of supporting the ecosystem around NumPy `2.0.0` support/wheels. I'll add a checklist item below for hearing objections/last-minute changes related to this acceleration of pace. Locally, this branch passed full suite alongside NumPy `2.0.0rc1` and the full doc build passed locally as well.

Backports Included (so far):

1. gh-20168
2. gh-20175
3. gh-20197
4. gh-20299
5. gh-20313
6. gh-20314
7. gh-20318
8. gh-20327
9. gh-20341
10. gh-20345
11. gh-20347 (partial cherry-pick because `pyproject.toml` was already updated on rel branch)
12. gh-20349
13. gh-20350
14. gh-20353
15. gh-20359
16. gh-20361 (only a direct pull-in of the license update from `main`, to mitigate risk)
17. gh-20362
18. gh-20364
19. gh-20337

TODO:

- [x] check the diff over (I'm working a bit fast)
- [x] hear out any objections to the accelerated pace, and/or additional backports needed to make folks happy about it
- [x] try the full wheel builds on this branch before it goes in, to increase likelihood of smooth release process

### CODE REVIEW

**Comment by ilayn** [comment]
I agree with the looming urgency of the matter but I'd say please take your time so that we don't create a self-inflicting stress for you. No problem with skipping RC2 for me.

**Comment by tylerjereddy** [comment]
Regular CI was passing, full wheel build test started.

**Comment by tylerjereddy** [comment]
All green with a few restarts. I'll let this sit overnight so there's a bit of time for comments or objections.

File: doc/source/release/1.13.0-notes.rst [hunk 1]
```diff
@@ -312,6 +326,7 @@ Issues closed for 1.13.0
 * `#19774 <https://github.com/scipy/scipy/issues/19774>`__: DOC: Detail what "concatenate" means in the context of \`spatial.transform.Rotation.concatenate\`
 * `#19799 <https://github.com/scipy/scipy/issues/19799>`__: DOC: array types: update array validation guidance
 * `#19813 <https://github.com/scipy/scipy/issues/19813>`__: BUG: typo in specfun.f?
+* `#19819 <https://github.com/scipy/scipy/issues/19819>`__: BUG: In RPFInterpolator, wrong warning message if degree=-1
```
Comment (rgommers) [inline doc/source/release/1.13.0-notes.rst:329]: very minor, typo: `RPFInterpolator` -> `RBFInterpolator` (I fixed the issue title itself just now)

File: doc/source/release/1.13.0-notes.rst [hunk 2]
```diff
@@ -51,6 +49,9 @@ New features
 - The Modified Akima Interpolation has been added to
   ``interpolate.Akima1DInterpolator``, available via the new ``method``
   argument.
+- New method ``BSpline.insert_knot`` inserts a knot into a ``BSpline` instance.
```
Comment (rgommers) [inline doc/source/release/1.13.0-notes.rst:52]: very minor: missing a second backtick behind `BSpline`

File: tools/openblas_support.py [hunk 3]
```diff
@@ -231,6 +231,26 @@ def get_members():
             yield member
 
     tarfileobj.extractall(target_path, members=get_members())
+    reformat_pkg_file(target_path=target_path)
+
+
+def reformat_pkg_file(target_path):
```
Comment (rgommers) [inline tools/openblas_support.py:237]: minor: if you're pushing another comment, perhaps add a comment that this is simple typo that is already fixed upstream in `openblas#4594`

**Review by rgommers — APPROVED** [review]
LGTM, thanks Tyler! I added a couple of very minor comments, but those seem optional to fix and don't even require retesting.

**Comment by tylerjereddy** [comment]
I addressed Ralf's comments and the docs build still passed after that.

I manually re-read the diff and didn't find anything accidentally pulled in. I did notice one new test that seems to use the "old" rather than newer/scoped NumPy random machinery, and one new error message code path with a typo in the message, but I don't think those merit additional attention for now.

**Comment by tylerjereddy** [comment]
alright, starting release process, what could possibly go wrong

**Pull request merged**